Mathlib/RingTheory/Valuation/ValuativeRel.lean

```diff
@@ -0,0 +1,589 @@
+/-
+Copyright (c) 2025 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Liu, Adam Topaz
+-/
+import Mathlib.RingTheory.Valuation.Basic
+import Mathlib.Data.NNReal.Defs
+import Mathlib.Topology.Defs.Filter
+
+/-!
+
+# Valuative Relations
+
+In this file we introduce a class called `ValuativeRel R` for a ring `R`.
+This bundles a relation `rel : R → R → Prop` on `R` which mimics a
+preorder on `R` arising from a valuation.
+We introduce the notation `x ≤ᵥ y` for this relation.
+
+Recall that the equivalence class of a valuation is *completely* characterized by
+such a preorder. Thus, we can think of `ValuativeRel R` as a way of
+saying that `R` is endowed with an equivalence class of a valuation.
+-/
+
+noncomputable section
+
+/-- The class `[ValuativeRel R]` class introduces an operator `x ≤ᵥ y : Prop` for `x y : R`
+which is the natural relation arising from an equivalence class of a valuation on `R`.
+More precisely, if v is a valuation on R then the associated relation is `x ≤ᵥ y ↔ v x ≤ v y`.
+Use this class to talk about the case where `R` is equipped with an equivalence class
+of valuations. -/
+class ValuativeRel (R : Type*) [CommRing R] where
```

Comment:
As mentioned in a private DM, I think that this class should *replace* `Valued`, lest we end up with two competing definitions.

Reply:
I agree we should do this refactor eventually, but this PR is already long enough. I have added a comment in the module docstrings of both `ValuativeRel` and `Valued` explaining the situation.